lib/asyn-thrdd.c

```diff
@@ -318,8 +318,14 @@ static void async_thrdd_destroy(struct Curl_easy *data)
 
     /* Release our reference to the data shared with the thread. */
     Curl_mutex_acquire(&addr->mutx);
-    thrdd->addr = NULL;
     --addr->ref_count;
+    /* we give up our reference to `addr`, so NULL our pointer.
+     * coverity analyses this as being a potential unsynched write,
+     * assuming two calls to this function could be invoked concurrently.
+     * Which they are never, as the transfer's side runs single-threaded.
+     * So assign only conditionally, hoping it silences coverity. */
+    if(thrdd->addr)
```

Comment:
Maybe just null it unconditionally? Surely the check here doesn't add much?

Reply:
Well, it had been an unconditional assignment to NULL that coverity complained about. =)

Reply:
oh right...

Reply:
So how about we just add the comment for educating the reader and we ignore the Coverity complaint?

Reply:
Done.